sdk/rust/oak/src/io/receiver.rs

```diff
@@ -66,14 +66,30 @@ impl<T: Decodable> Receiver<T> {
     }
 
     /// Wait for a value to be available from this receiver.
+    ///
+    /// May return the following error statuses:
+    /// - `Err(OakStatus::ErrTerminated)`: if the Oak Runtime is terminating.
+    /// - `Err(OakStatus::ErrInternal)`: if cannot extract the `ChannelReadStatus` form the buffer
+    ///   from `wait_on_channels`.
     #[allow(clippy::trivially_copy_pass_by_ref)]
-    pub fn wait(&self) -> Result<(), OakStatus> {
+    pub fn wait(&self) -> Result<ChannelReadStatus, OakStatus> {
         // TODO(#500): Consider creating the handle notification space once and for all in `new`.
         let read_handles = vec![self.handle];
         let mut space = crate::new_handle_space(&read_handles);
         crate::prep_handle_space(&mut space);
         let status =
             unsafe { oak_abi::wait_on_channels(space.as_mut_ptr(), read_handles.len() as u32) };
-        crate::result_from_status(status as i32, ())
+
+        match space
+            .get(oak_abi::SPACE_BYTES_PER_HANDLE - 1)
```

Comment:
Trying to get the byte by addressing it from the end seems harder to follow and more fragile in case the layout changes (assuming it's more likely we will add things to the end of this representation). In fact I think perhaps we should consider a proper struct (perhaps tagged with `repr(c)`) to make sure that `new_handle_space` and this logic are in sync and binary-compatible (also ideally with the Runtime side of things, but I'm not sure that's feasible. Or we should just bite the bullet and separate inputs and outputs as per https://github.com/project-oak/oak/issues/918. WDYT? cc @daviddrysdale